misk/web/@misk/components/src/components/TopbarComponentDimensionAware.tsx

```diff
@@ -0,0 +1,388 @@
+    console.log(this.props.navbarItems)
+    return [this.renderNavbarEnvironment(environment)].concat(
+      this.props.navbarItems
+        ? this.props.navbarItems.map(item => (
+            <MiskNavbarHeading>
+              <TextHTMLOrElementComponent content={item} />
+            </MiskNavbarHeading>
+          ))
+        : []
+    )
+  }
+
+  private renderBanner(
+    status?: string | Element | JSX.Element,
+    environment?: Environment,
+    environmentBannerVisible: Environment[] = [
+      Environment.DEVELOPMENT,
+      Environment.STAGING,
+      Environment.TESTING
+    ]
+  ) {
+    if (environmentBannerVisible.includes(environment)) {
+      if (typeof status === "string" && !status.startsWith("<")) {
+        console.log(`[STATUS] ${status}`)
+      }
+      return (
+        <MiskBanner color={environmentToColor(environment)}>
+          <ResponsiveContainer>
+            <FlexContainer>
+              <TextHTMLOrElementComponent content={status} />
+            </FlexContainer>
+          </ResponsiveContainer>
+        </MiskBanner>
+      )
+    } else {
+      return <div />
+    }
+  }
+
+  private renderNavbarHome(homeName: string, homeUrl: string) {
+    if (homeName && homeUrl) {
+      return (
+        <MiskLinkHome to={homeUrl}>
+          <MiskNavbarHeading>{homeName}</MiskNavbarHeading>
+        </MiskLinkHome>
+      )
+    } else if (homeName) {
+      return <MiskNavbarHeading>{homeName}</MiskNavbarHeading>
+    } else {
+      return <MiskNavbarHeading>Misk</MiskNavbarHeading>
+    }
+  }
+
+  private renderNavbarEnvironment(
+    environment?: Environment,
+    environmentBannerVisible: Environment[] = [
+      Environment.DEVELOPMENT,
+      Environment.STAGING,
+      Environment.TESTING
+    ]
+  ) {
+    if (environmentBannerVisible.includes(environment)) {
+      return (
+        <MiskNavbarHeadingEnvironment color={environmentToColor(environment)}>
+          {environment}
+        </MiskNavbarHeadingEnvironment>
+      )
+    } else {
+      return <div />
+    }
+  }
+
+  private renderMenuCategories(links: IDashboardTab[]) {
+    const categories: Array<[string, IDashboardTab[]]> = Object.entries(
+      groupBy(links, "category")
+    )
+    return categories.map(([categoryName, categoryLinks]) =>
+      this.renderMenuCategory(categoryName, categoryLinks)
+    )
+  }
+
+  private renderMenuCategory(
+    categoryName: string,
+    categoryLinks: IDashboardTab[]
+  ) {
+    const sortedCategoryLinks = sortBy(categoryLinks, "name").filter(
+      (link: IDashboardTab) => link.category !== ""
+    )
+    return (
+      <div>
+        <MiskMenuCategory>{categoryName}</MiskMenuCategory>
+        <MiskMenuDivider />
+        <MiskMenuLinks>
+          {sortedCategoryLinks.map((link: IDashboardTab) => (
+            <MiskMenuLink
+              key={link.slug}
+              onClick={this.handleClick}
+              to={link.url_path_prefix}
+            >
```

Comment:
nit: is it intentional to have this on its own line?

Reply:
`jsx-alignment` rule in TSLint https://github.com/palantir/tslint-react